File: backend/src/main/java/com/backend/ord/domain/entities/BankGroup.java
package com.backend.ord.domain.entities;

import com.backend.ord.domain.entities.abstracts.EntityBase;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Entity
@Table(name = "bank_groups")
public class BankGroup extends EntityBase {
    @Column(name = "name", nullable = false, length = 64)
    private String name;

    @Column(name = "color", nullable = false, length = 6)
    private String color;

    public BankGroup(String name, String color) {
        this.name = name;
        this.color = color;
    }

    public BankGroup() {
    }

    public static BankGroupBuilder builder() {
        return new BankGroupBuilder();
    }

    public String getName() {
        return this.name;
    }

    public String getColor() {
        return this.color;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public static class BankGroupBuilder {
        private String name;
        private String color;

        BankGroupBuilder() {
        }

        public BankGroupBuilder name(String name) {
            this.name = name;
            return this;
        }

        public BankGroupBuilder color(String color) {
            this.color = color;
            return this;
        }

        public BankGroup build() {
            return new BankGroup(this.name, this.color);
        }

        public String toString() {
            return "BankGroup.BankGroupBuilder(name=" + this.name + ", color=" + this.color + ")";
        }
    }
}

File: backend/src/main/kotlin/com/backend/ord/domain/entities/BankGroup.kt
package com.backend.ord.domain.entities

import com.backend.ord.domain.entities.interfaces.IdentifiableUserResource
import jakarta.persistence.*
import jakarta.validation.constraints.Size
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.*

@Entity
@Table(name = "bank_groups")
class BankGroup(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    override var id: UUID = UUID.randomUUID(),

    @field:Size(max = 64)
    @Column(name = "name", nullable = false)
    var name: String,

    @field:Size(max = 7)
    @Column(name = "color", nullable = false)
    var color: String,

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    override var user: User,

    @Column(name = "created_at", nullable = false, updatable = false)
    @CreationTimestamp
    var createdAt: Instant = Instant.now(),

    @Column(name = "updated_at", nullable = false)
    @UpdateTimestamp
    var updatedAt: Instant = Instant.now()
) : IdentifiableUserResource